src/main/java/menion/android/whereyougo/gui/extension/activity/CustomActivity.java

```diff
@@ -253,14 +253,34 @@ private void unbindDrawables(View view) {
     }
 
     public static void setLocale(Activity activity, String languageCode) {
-        String lang = languageCode;
+        Locale locale;
+        String lang;
+        Resources resources = activity.getResources();
+        Configuration config = resources.getConfiguration();
+
+        // we need to get language and region first even for default to correctly initialize Locale
         if (languageCode.equals("default")) {
-            lang = Locale.getDefault().getLanguage();
+            lang = Locale.getDefault().toString();
+        } else {
+            lang = new Locale(languageCode).toString();
+        }
+
+        // need to parse language codes with dash and hyphen
+        String[] loc = lang.split("[-_]");
+        if (loc.length == 1) {
+            if (loc[0].equals("pt")) {
+                // nasty hotfix for Portuguese language selected without regional code
+                locale = new Locale(loc[0], loc[0]);
```

Comment:
Let's do it the way around - rename the language `pt_PT` to `pt`.
I created Issue #342 for it.
After that these lines are not more necessary.